domo_blueprints/upload_csv_to_dataset.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import argparse
+from pydomo import Domo
+import shipyard_utils as shipyard
+import pandas as pd
+try:
+    import errors as ec
+except BaseException:
+    from . import errors as ec
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--client-id', dest='client_id', required=True)
+    parser.add_argument('--secret-key', dest='secret_key', required=True)
+    parser.add_argument('--file-to-load', dest='file_to_load', required=True)
+    parser.add_argument('--dataset-name', dest='dataset_name', required=True)
+    parser.add_argument('--dataset-description', dest = 'dataset_description', required=False)
+    args = parser.parse_args()
+
+    return args
+
+
+def dataset_exists(datasets, dataset_name):
+    return datasets.name.str.contains(dataset_name).any()
+
+
+def get_csv(file_name:str):
+    """
+    Reads in a local file as a csv
+    @params: 
+        file_name : str - the name of the file to be loaded (provided by the --file-to-load flag)
+    """
+    try:
+        df = pd.read_csv(file_name)
+        return df
+    except Exception as e:
+        print(f"File {file_name} was not found, ensure that the name of the file is entered correctly")
+        sys.exit(ec.EXIT_CODE_FILE_NOT_FOUND)
+
+
+def upload_csv(dataframe:pd.DataFrame, domo_instance, dataset_name:str, dataset_description = None):
+    """
+    Uploads a pandas dataframe as a dataset in Domo.
+    @params:
+        dataframe : DataFrame - The pandas dataframe desired to load into Domo
+        domo_instance : Domo - The Domo instance you are connected to 
+        dataset_namme : str - The name that the dataset will be given in Domo
+        dataset_description : str | None - Optional argument that provides a description for the dataset in Domo
+    """
+    try:
+        if dataset_description is None:
+            df_dataset = domo_instance.ds_create(dataframe,dataset_name)
+        else:
+            df_dataset = domo_instance.ds_create(dataframe,dataset_name,dataset_description)
+        print(f"Successfully created dataset {dataset_name} with id {df_dataset}")
+        return df_dataset
+    except Exception as e:
+        print(f"Error in uploading the csv to domo")
+        print(e)
+        sys.exit(ec.EXIT_CODE_UNKNOWN_ERROR)
```

Comment:
If you provide a dataset that already exists but wasn't created with the API, you get this error:
`Exception: Error uploading Data Part on Execution 70 on Stream 12: {"status":400,"statusReason":"Bad Request","message":"Only DataSets created with the API can be updated via APIs.","toe":"4OUM9IBSTB-E3B3K-V1HVP"}`
We should have a separate way to handle for this error, rather than letting it be "unknown".